compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -851,6 +851,31 @@ static void PrintMethodHandlerClass(const ServiceDescriptor* service,
   p->Print("}\n\n");
 }
 
+static void PrintGetServiceDescriptorMethod(const ServiceDescriptor* service,
+                                   map<string, string>* vars,
+                                   Printer* p,
+                                   bool generate_nano) {
+  (*vars)["service_name"] = service->name();
+  p->Print(
+      *vars,
+      "public static $ServiceDescriptor$ getServiceDescriptor() {\n");
```

Comment:
Do we want an `@ExperimentalApi` annotation here? I know we'll want at method to get this information, but I don't know if we're going to want it to be non-static, or similar. I guess worst-case if it wasn't experimental it isn't hard to maintain compatability with this method.